osu.Game/Beatmaps/IBeatmapSetInfo.cs

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections.Generic;
+using osu.Game.Database;
+
+#nullable enable
+
+namespace osu.Game.Beatmaps
+{
+    /// <summary>
+    /// A representation of a collection of beatmap difficulties, generally packaged as an ".osz" archive.
+    /// </summary>
+    public interface IBeatmapSetInfo : IHasOnlineID
+    {
+        /// <summary>
+        /// The date when this beatmap was imported.
+        /// </summary>
+        DateTimeOffset DateAdded { get; }
+
+        /// <summary>
+        /// The best-effort metadata representing this set. In the case metadata differs between contained beatmaps, one is arbitrarily chosen.
+        /// </summary>
+        IBeatmapMetadataInfo? Metadata { get; }
+
+        /// <summary>
+        /// All beatmaps contained in this set.
+        /// </summary>
+        IEnumerable<IBeatmapInfo> Beatmaps { get; }
```

Comment:
Shouldn't this be named `BeatmapInfos` to be in line with the new naming?

Reply:
Very unsure about that.. just because `BeatmapInfos` sounds weird.